code/modules/reagents/reagent_containers/borghydro.dm

```diff
@@ -100,11 +100,8 @@ Borg Hypospray
 	return
 
 /obj/item/weapon/reagent_containers/borghypo/attack_self(mob/user)
-	mode++
-	if(mode > reagent_list.len)
-		mode = 1
+	mode = modes[input(user, "What reagent do you want to dispense?") as anything in reagent_ids]

```

Comment:
move the input statement out of the ```modes[]``` then swap anything for null|anything and then do a check for ```if(var_you_put_input_in)``` so that users can simply cancel out, instead of having to pick the same chemical.